### PR TITLE
JP-2071: Update ramp_fit to toggle one-group slope calculations

### LINKED ISSUE #6013: Add ramp_fit parameter to turn on/off one-group slope calculations
_Issue [JP-2071](https://jira.stsci.edu/browse/JP-2071) was created on JIRA by [Michael Regan](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mregan):_

Currently, the ramp fitting step will fit ramps with only a single useable group using the difference between the first group and the superbias. This assumes that the superbias reference file used darks taken at the same temperature as the current observations. Because there will be secular drifts in temperature over the life of the observatory (especially during commissioning), there can be a mismatch in temperature which will lead to the wrong slope.
It would be best to have a parameter value that can turn this off and on with the default being off until this is tested with on-orbit observations.

### PR DESCRIPTION
**Pull request by kmacdonald-stsci**
Closes #6013 
Resolves [JP-2071](https://jira.stsci.edu/browse/JP-2071)

**Description**

A `suppress_one_group` boolean variable has been added as a parameter to the ramp fitting step class.

When true (the default) saturated ramps with only the 0th group being good will be treated as completely saturated ramps. The calculations will be done as if all calculations are saturated, but the DQ flags will be set as if the first group is good, i.e., the pixel DQ will not have DO_NOT_USE set.

When false saturated ramps with only the 0th group being good will be treated as the normal special case of having only one good group.

Many unit tests are expected to fail until the corresponding PR in STCAL has been merged.

Checklist
- [x] Tests
- [ ] Documentation
- [x] Change log
- [x] Milestone
- [x] Label(s)

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -48,6 +48,9 @@ ramp_fitting
   the various flavors of variance and ERR stored in the output
   products [#6715]
 
+- Adding feature to turn off calculations of ramps with good 0th group,
+  but all other groups are saturated. [#]
```
Comment (hbushouse) [inline CHANGES.rst:52]: add 6737 to PR number

File: jwst/ramp_fitting/ramp_fit_step.py [hunk 2]
```diff
@@ -240,7 +241,11 @@ def process(self, input):
             image_info, integ_info, opt_info, gls_opt_model = ramp_fit.ramp_fit(
                 input_model, buffsize,
                 self.save_opt, readnoise_2d, gain_2d, self.algorithm,
-                self.weighting, max_cores, dqflags.pixel)
+                self.weighting, max_cores, dqflags.pixel, suppress_one_group=self.suppress_one_group)
+
+            print("-" * 80)
```
Comment (hbushouse) [inline jwst/ramp_fitting/ramp_fit_step.py:246]: don't want to leave these in the final product

**Review by hbushouse — COMMENTED** [review]
Looks OK. Just needs a couple minor updates before finalizing.

File: jwst/ramp_fitting/tests/test_ramp_fit_step.py [hunk 3]
```diff
@@ -246,3 +247,232 @@ def test_int_times2(generate_miri_reffiles, setup_inputs):
...
+        override_gain=gmodel,
+        override_readnoise=rnModel,
+        suppress_one_group=suppress,
+        maximum_cores="none")
+
+    return slopes, cube_model, dims
+
+
+def test_one_group_not_suppressed_one_integration(setup_inputs):
+    slopes, cube, dims = one_group_suppressed(1, False, setup_inputs)
```
Comment (dmggh) [inline jwst/ramp_fitting/tests/test_ramp_fit_step.py:316]: Include short description of test purpose (also for later tests)

**Review by dmggh — APPROVED** [review]
Looks good to me.  Just a couple of minor docs comments.

File: CHANGES.rst (hunk 1 above)
Comment (kmacdonald-stsci) [reply CHANGES.rst:52]: I corrected this.

File: jwst/ramp_fitting/tests/test_ramp_fit_step.py (hunk 3 above)
Comment (kmacdonald-stsci) [reply jwst/ramp_fitting/tests/test_ramp_fit_step.py:316]: I included short descriptions for the tests.

File: CHANGES.rst [hunk 4]
```diff
@@ -46,7 +46,10 @@ ramp_fitting
 
 - Updated step docs to clarify exactly what calculations are used for
   the various flavors of variance and ERR stored in the output
-  products [#6715]
+  products [#6737]
```
Comment (hbushouse) [inline CHANGES.rst:49]: The 6737 reference needs to go down into line 52 (the entry for this PR). The reference to 6715 needs to stay as is (that was my PR for doc updates).

File: jwst/ramp_fitting/ramp_fit_step.py (hunk 2 above)
Comment (kmacdonald-stsci) [reply jwst/ramp_fitting/ramp_fit_step.py]: I removed these.

**Review by hbushouse — APPROVED** [review]
LGTM.  I guess we have to merge the stcal PR first and then this one?

**Comment by hbushouse** [comment]
It occurs to me that the docs for the ramp_fit step will also need to be updated to list this new argument and what it does.

**Comment by kmacdonald-stsci** [comment]
I updated the `arguments.rst` documentation for ramp fitting.

**Comment by hbushouse** [comment]
Failure in CI for oldest dependencies is due to the use of stcal 0.6.0, which does not support the new "suppress_one_group" step param. It is supported in stcal 0.6.1, which is used in the other passing CI tests.

**Pull request merged**